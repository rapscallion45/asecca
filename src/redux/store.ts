import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
