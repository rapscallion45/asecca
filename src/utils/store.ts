/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
