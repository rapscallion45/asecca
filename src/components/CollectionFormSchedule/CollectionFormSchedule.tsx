import { FC, useEffect, useCallback, ChangeEvent } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import dayjs, { Dayjs } from 'dayjs';
import { AppState, AppDispatch } from '@/redux/store';
import {
  Card,
  CardHeader,
  CardContent,
  Box,
  Button,
  TextField,
  Stack,
  Skeleton,
} from '@mui/material';
import { DatePicker } from '@mui/x-date-pickers/DatePicker';
import { TimeField } from '@mui/x-date-pickers/TimeField';
import { LoadingButton } from '@mui/lab';
import {
  saveByCollectionId as saveScheduleByCollectionId,
  editSchedule,
  resetSchedule,
  fetchByCollectionId,
} from '@/redux/slices/collectionFormScheduleSlice';

/**
 * Collection Form Schedule Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 *
 * @typedef ICollectionFormScheduleProps
 * @prop {string} collectionId - ID string of Collection for data API call
 */
interface ICollectionFormScheduleProps {
  collectionId: string;
}

/**
 * Collection Form Schedule
 *
 * Presents the Collection Form Schedule form to the user, populated with data
 * fetched from API: /api/collection/schedule/api/get
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 *
 * @component
 * @param {ICollectionFormScheduleProps} props - component props
 * @returns {FC} - collection form schedule form functional component
 */
const CollectionFormSchedule: FC<ICollectionFormScheduleProps> = (props) => {
  const { collectionId } = props;

  /* shorthand helper for dispatching redux actions */
  const dispatch = useDispatch<AppDispatch>();

  /* get collection form schedule data held in redux state */
  const {
    data: scheduleData,
    loading,
    error,
    saving,
    edited,
  } = useSelector((state: AppState) => state.collectionFormSchedule);

  /* on change of Collection ID, fetch data from API */
  useEffect(() => {
    dispatch(fetchByCollectionId({ collectionId }));
  }, [collectionId, dispatch]);

  /**
   * Handles the editing of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.13
   *
   * @method
   * @param {string} value - form value to be updated
   * @param {string} itemKey - form item key to be updated
   */
  const handleEdit = (value: string, itemKey: string) => {
    dispatch(editSchedule({ itemKey, value }));
  };

  /**
   * Handles the resetting of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.13
   *
   * @method
   */
  const handleReset = useCallback(() => {
    dispatch(resetSchedule());
  }, [dispatch]);

  /**
   * Handles the saving of the form data
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.13
   *
   * @method
   */
  const handleSave = useCallback(() => {
    dispatch(
      saveScheduleByCollectionId({
        data: {
          collection: collectionId,
          preferred_date: scheduleData?.preferred_date,
          preferred_time: scheduleData?.preferred_time,
          notes: scheduleData?.notes,
        },
      })
    );
  }, [collectionId, scheduleData, dispatch]);

  return (
    <Card>
      <CardHeader title="Schedule" />
      <CardContent sx={{ pt: 0 }}>
        {!loading && (
          <Box
            component="form"
            sx={{
              '& .MuiTextField-root': { my: 1 },
            }}
            noValidate
            autoComplete="off"
            display="flex"
            flexDirection="column"
          >
            <Box display="flex">
              <DatePicker
                label="Preferred Date"
                value={dayjs(scheduleData?.preferred_date || '2024/09/08')}
                onChange={(newValue: Dayjs | null) =>
                  handleEdit(
                    newValue ? newValue.format('YYYY/MM/DD') : '',
                    'preferred_date'
                  )
                }
                sx={{ mr: 2 }}
              />
              <TimeField
                label="Preferred Time"
                value={dayjs(
                  `${scheduleData?.preferred_date || '2024/09/08'}T${
                    scheduleData?.preferred_time || '09:30'
                  }`
                )}
                onChange={(newValue: Dayjs | null) =>
                  handleEdit(
                    newValue ? newValue.format('hh:mm') : '',
                    'preferred_time'
                  )
                }
                color="secondary"
              />
            </Box>
            <TextField
              id="schedule-notes-multiline"
              label="Notes"
              multiline
              maxRows={4}
              variant="standard"
              value={scheduleData?.notes || ''}
              onChange={(event: ChangeEvent<HTMLInputElement>) => {
                handleEdit(event.target.value, 'notes');
              }}
              color="secondary"
            />
          </Box>
        )}
        {loading && (
          <Stack spacing={1}>
            {/* display loading skeleton */}
            <Stack spacing={2} direction="row">
              <Skeleton variant="rectangular" width={310} height={60} />
              <Skeleton variant="rectangular" width={310} height={60} />
            </Stack>
            <Skeleton variant="rectangular" width="100%" height={40} />
          </Stack>
        )}
        <Box
          sx={{
            marginTop: 2,
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'left',
          }}
        >
          <Button
            color="secondary"
            variant="outlined"
            onClick={handleReset}
            disabled={saving || loading || !edited || Boolean(error)}
            sx={{ backgroundColor: 'common.white' }}
          >
            Cancel
          </Button>
          <LoadingButton
            color="secondary"
            variant="contained"
            onClick={handleSave}
            disabled={saving || loading || !edited || Boolean(error)}
            loading={saving}
            sx={{ ml: 2 }}
          >
            Commit
          </LoadingButton>
        </Box>
      </CardContent>
    </Card>
  );
};

export default CollectionFormSchedule;
