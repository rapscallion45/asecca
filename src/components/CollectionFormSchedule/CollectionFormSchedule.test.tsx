import React from 'react';
import { Provider } from 'react-redux';
// import {
//   fireEvent,
//   render,
//   screen,
//   waitFor,
//   act,
// } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import store from '@/redux/store';
import CollectionFormSchedule from './CollectionFormSchedule';
// import collectionFormScheduleDataMock from '../../../__mocks__/CollectionForm/collectionFormScheduleDataMock';

/* default test query ID */
const query: string = '66135000001760012';

/**
 * Collection Form Schedule Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.22
 */
describe('Collection Form Schedule', () => {
  it('Renders correctly', async () => {
    /** Arrange */
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Provider store={store}>
          <LocalizationProvider dateAdapter={AdapterDayjs}>
            <CollectionFormSchedule collectionId={query} />
          </LocalizationProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // describe('Form Fields', () => {
  //   it('Should render all required New Contact data fields', async () => {
  //     /** Arrange */
  //     /** Act - render the test components */
  //     render(
  //       <Provider store={store}>
  //         <LocalizationProvider dateAdapter={AdapterDayjs}>
  //           <CollectionFormSchedule collectionId={query} />
  //         </LocalizationProvider>
  //       </Provider>
  //     );

  //     /** Assert - ensure all fields are present */
  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Preferred Date')).toBeInTheDocument();
  //       expect(screen.getByLabelText('Preferred Time')).toBeInTheDocument();
  //       expect(screen.getByLabelText('Notes')).toBeInTheDocument();
  //     });
  //   });
  // });

  // describe('Form Callbacks', () => {
  //   beforeAll(() => jest.spyOn(window, 'fetch'));

  //   it('Should call API with inputted values when values updated and save button clicked', async () => {
  //     /** Arrange */
  //     const testPreferredDate = '10/27/2024';
  //     const formattedTestDate = '2024/10/27';
  //     const testPreferredTime = '02:30 PM';
  //     const formattedPreferredTime = '02:30';
  //     const testNotes = 'This is a test notes string.';
  //     act(() => {
  //       store.dispatch(
  //         fetchCollectionFormScheduleByCollectionId({
  //           collectionId: query,
  //         })
  //       );
  //     });

  //     /** Assert - get API to have been called */
  //     await waitFor(() => {
  //       expect(window.fetch).toHaveBeenCalledTimes(1);
  //     });
  //     expect(window.fetch).toHaveBeenCalledWith(
  //       `/api/collection/schedule/api/get?collection=${query}`,
  //       expect.objectContaining({
  //         method: 'GET',
  //         headers: { 'Content-Type': 'application/json' },
  //       })
  //     );

  //     /** Act - render the test components */
  //     render(
  //       <Provider store={store}>
  //         <LocalizationProvider dateAdapter={AdapterDayjs}>
  //           <CollectionFormSchedule collectionId={query} />
  //         </LocalizationProvider>
  //       </Provider>
  //     );

  //     /** Assert - ensure data is loaded */
  //     await waitFor(() => {
  //       expect(
  //         screen.getByDisplayValue(
  //           collectionFormScheduleDataMock.notes as string
  //         )
  //       ).toBeInTheDocument();
  //     });
  //     /** Act - update field values and click Save button */
  //     act(() => {
  //       fireEvent(
  //         screen.getByLabelText('Preferred Date'),
  //         new MouseEvent('click', {
  //           bubbles: true,
  //           cancelable: true,
  //         })
  //       );
  //       fireEvent.change(screen.getByPlaceholderText('MM/DD/YYYY'), {
  //         target: { value: testPreferredDate },
  //       });
  //     });
  //     act(() => {
  //       fireEvent(
  //         screen.getByLabelText('Preferred Time'),
  //         new MouseEvent('click', {
  //           bubbles: true,
  //           cancelable: true,
  //         })
  //       );
  //       fireEvent.change(screen.getByPlaceholderText('hh:mm aa'), {
  //         target: { value: testPreferredTime },
  //       });
  //     });
  //     act(() => {
  //       fireEvent(
  //         screen.getByLabelText('Notes'),
  //         new MouseEvent('click', {
  //           bubbles: true,
  //           cancelable: true,
  //         })
  //       );
  //       fireEvent.change(screen.getByLabelText('Notes'), {
  //         target: { value: testNotes },
  //       });
  //     });
  //     /** Act - click Commit button */
  //     act(() => {
  //       fireEvent(
  //         screen.getByText('Commit'),
  //         new MouseEvent('click', {
  //           bubbles: true,
  //           cancelable: true,
  //         })
  //       );
  //     });
  //     /** Assert - save API to have been called with data from user input */
  //     await waitFor(() => {
  //       expect(window.fetch).toHaveBeenCalledTimes(2);
  //     });
  //     expect(window.fetch).toHaveBeenCalledWith(
  //       '/api/collection/schedule/api/set',
  //       expect.objectContaining({
  //         method: 'POST',
  //         headers: { 'Content-Type': 'application/json' },
  //         body: JSON.stringify({
  //           collection: query,
  //           preferred_date: formattedTestDate,
  //           preferred_time: formattedPreferredTime,
  //           notes: testNotes,
  //         }),
  //       })
  //     );
  //   });
  // });
});
