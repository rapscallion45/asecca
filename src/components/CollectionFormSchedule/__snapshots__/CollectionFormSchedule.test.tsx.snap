// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Form Schedule Renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
      >
        Schedule
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root css-1hizqsf-MuiCardContent-root"
  >
    <form
      autoComplete="off"
      className="MuiBox-root css-1b7bzod"
      noValidate={true}
    >
      <div
        className="MuiBox-root css-k008qs"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-5o1wk7-MuiFormControl-root-MuiTextField-root"
          onClick={[Function]}
          onMouseUp={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Preferred Date
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              inputMode="text"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              placeholder="MM/DD/YYYY"
              readOnly={false}
              required={false}
              type="text"
              value="09/08/2024"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is Sep 8, 2024"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-14lo706"
              >
                <span>
                  Preferred Date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root"
          onClick={[Function]}
          onMouseUp={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-179z3gj-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor=":r2:"
            id=":r2:-label"
          >
            Preferred Time
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r2:"
              inputMode="text"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              placeholder="hh:mm aa"
              readOnly={false}
              required={false}
              type="text"
              value="09:30 AM"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-14lo706"
              >
                <span>
                  Preferred Time
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-t7bnno-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="schedule-notes-multiline"
          id="schedule-notes-multiline-label"
        >
          Notes
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInputBase-formControl MuiInputBase-multiline css-1w7ptt9-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
            disabled={false}
            id="schedule-notes-multiline"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            style={
              {
                "height": 0,
                "overflow": undefined,
              }
            }
            value=""
          />
          <textarea
            aria-hidden={true}
            className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
            readOnly={true}
            style={
              {
                "height": 0,
                "left": 0,
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "transform": "translateZ(0)",
                "visibility": "hidden",
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
    </form>
    <div
      className="MuiBox-root css-kh5ii8"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1ex1hsz-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Cancel
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1fddvdp-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        disabled={true}
        id=":r4:"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        Commit
      </button>
    </div>
  </div>
</div>
`;
